CMakeLists.txt

```diff
@@ -339,8 +342,21 @@ if(EMSCRIPTEN)
     add_compile_flag("-sDISABLE_EXCEPTION_CATCHING=0")
     add_link_flag("-sDISABLE_EXCEPTION_CATCHING=0")
   endif()
-  # make the tools immediately usable on Node.js
-  add_link_flag("-sNODERAWFS")
+  # make the tools immediately usable on Node.js, unless building for the browser
+  if (NOT BUILD_FOR_BROWSER)
+    add_link_flag("-sNODERAWFS")
+  endif()
+  if(BUILD_FOR_BROWSER)
+    add_link_flag("-sENVIRONMENT=web,worker")
+    add_link_flag("-sINVOKE_RUN=0")
+    add_link_flag("-sNODERAWFS=0")
+    add_link_flag("-sMODULARIZE=1")
+    add_link_flag("-sEXPORT_ES6")
+    add_link_flag("-sFILESYSTEM=1")
+    add_link_flag("-sFORCE_FILESYSTEM")
+    add_link_flag("-sSINGLE_FILE=0")
```

Comment:
nits:

You don't need `-sSINGLE_FILE=0` or `-sNODERAWFS=0` since they are the default.

Also, you can drop the `=1` from MODULARIZE and FILESYSTEM

Reply:
Addressed in https://github.com/WebAssembly/binaryen/pull/6131/commits/246ea0ec125f603cbbbe599b114ae0ffb02524b9.